stats/cloud/client.go

```diff
@@ -94,6 +98,11 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 		}
 	}
 
+	// Set session token for POST request, if not existed yet.
+	if req.Method == http.MethodPost && req.Header.Get(k6SessionTokenHeader) == "" {
+		req.Header.Set(k6SessionTokenHeader, generateSessionToken())
```

Comment:
This seems wrong - we're going to generate a different random token for each request. Instead, we need the same token across all requests, i.e. we need to generate the session token and save it in `NewClient()` and then always reuse it.

Reply:
Are you sure? I think the token is reused for retry request only, no?

Reply:
In an Slack discussion @cuonglm convinced me that a per-request idempotency key (for non-`HEAD`/non-`GET` requests) would be the better solution for retrying requests, particularly in cases where we have to call the same endpoint multiple times (like metric pushing). 

But at the same time, a unique session key that's the same for all requests (`HEAD`/`GET` included) in a single test run could also be useful. In this case the backend could use either, but the session ID would be more useful for things like tracing and logging, if nothing else... So, I suggest we add both :smile:

Reply:
@na-- Implemented in latest commits, please take a look.